sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/DotCoverReportParserTest.java

```diff
@@ -55,17 +53,56 @@ public void prepare() {
 
   @Test
   public void no_title() {
-    thrown.expect(IllegalArgumentException.class);
-    thrown.expectMessage("The report contents does not match the following regular expression: .*?<title>(.*?)</title>.*");
-    new DotCoverReportParser(alwaysTrue).accept(new File("src/test/resources/dotcover/no_title.html"), mock(Coverage.class));
+    DotCoverReportParser parser = new DotCoverReportParser(alwaysTrue);
```

Comment:
Codesmell: SL complains about merging these 3 tests to a single, parametrized one.

Reply:
I know, but I'd have to update to JUnit5 for this and I don't want to do it in this PR.

The JUnit4 way of dealing with parameterized tests is depressing. I opened #4223 for updating to JUnit5.

Reply:
(tried in some PR before to update the JUnit version, either for UTs or ITs, and there was a problem at that point - maybe a small one... so it's better to have a separate ticket)